### ISSUE TITLE
The face skin is deteriorated after cloaking with sign similar to scars or imperfections

### ISSUE THREAD

**Issue by panderlad** [Shawn-Shan/fawkes#118]
This doesn't with all photos. I tried 8 photos and 3 are still acceptable but 5 looks bad after cloaking.

On your website you show that the images doesn't have any difference notable to the eyes but all 8 photos are notable different, how is possible?

**Comment by arthurlataks** [comment]
Yes, can confirm, experiencing same thing.

**Comment by xirkus** [comment]
Same here. Is there a way to control the resolution of the generated file? Starting out with a 300kb+ file reduces to 65kb.

**Comment by panderlad** [comment]
It's not about the image resolution the problem, is the algorithm that creates imperfections on the skin of the face

-------- Original Message --------

**Comment by MarkSCQ** [comment]
Same, I think the processed images should be aimed at being captured by Scrapy bots. Naked eyes can detect the differences between processed images and original images when setting strong perturbations, like mid or high modes in V0.3.2.

**Comment by xirkus** [comment]
Unfortunately, I still see significant perturbations on the default setting (which I believe is low/min).

**Comment by panderlad** [comment]
Same degradation thing with me in default settings.

-------- Original Message --------

**Comment by MarkSCQ** [comment]
> Unfortunately, I still see significant perturbations on the default setting (which I believe is low/min).
Maybe I should get another pair of glasses O_O

**Comment by xirkus** [comment]
@MarkSCQ https://dev.scrb.ai/#example

**Comment by MarkSCQ** [comment]
> @MarkSCQ https://dev.scrb.ai/#example

This is processed by fawkes? These perturbations are more clear.

I have some images processed last week. I used custom mode DSSIM=0.004, MaxIteration=120, LearningRate=40
The upper one: 86.2kb 261X281, the bottom one: 85.8kb 234X242. 

![image](https://user-images.githubusercontent.com/22400370/114593073-88afe180-9c40-11eb-897e-925f17954660.png)

**Comment by xirkus** [comment]
@MarkSCQ Interesting. Do you know what the impact is on using those custom settings on the level of protection and the computational protection cost for generating these images?

**Comment by xirkus** [comment]
@MarkSCQ Also, have you done any comparisons with fawkes 0.3.1 vs. 1.0.2 with the same settings?

**Comment by MarkSCQ** [comment]
@xirkus 
As far as I know, the larger DSSIM will make stronger perturbations, but they(perturbations) may distort the origin image at the same time. And Fawkes is used under clean label attack which take perturbed images as training set. 

Sometimes the higher DSSIM images cannot be recognized successfully if using the modified images as testing set (not clean label attack scenario). 

The custom mode is similar to the origin modes. It just replaces the original ones with custom DSSIM, Max Iterations and Learning Rate. I cannot come up with the idea of impacts behind Max Iterations and Learning Rate(I remember Fawkes team mentioned some basic idea of them in V0.3.2 code using some comments like Max Iterations: times of computation of Gradient Descent...), and it's beyond my abilities, I am not a math expert. As for DSSIM, from the paper and code V0.3.2, it obviously controls the overall performance of Fawkes.

I noticed the changes in source code, but I did not make the comparison about the two versions, as mentioned in #124 I don't think this is necessary currently.

What is computational protection cost? Do you mean the time of processing images or probabilities from classifier? 

If I knew the impacts of them, this would definitely be another paper hhhhh

**Comment by xirkus** [comment]
@MarkSCQ may want to have a look at issue [125](https://github.com/Shawn-Shan/fawkes/issues/125)
